core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/PaginatorUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.util;
+
+import java.util.Collection;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@SdkInternalApi
```

Comment:
Should this be protected? It's imported by the service modules right?

Reply:
yup, good call. I will update it and move it outside of internal package.